rollbar-java/src/main/java/com/rollbar/notifier/Rollbar.java

```diff
@@ -1,4 +1,633 @@
 package com.rollbar.notifier;
 
+import com.rollbar.api.payload.Payload;
+import com.rollbar.api.payload.data.Data;
+import com.rollbar.api.payload.data.Level;
+import com.rollbar.notifier.config.Config;
+import com.rollbar.notifier.uncaughtexception.RollbarUncaughtExceptionHandler;
+import com.rollbar.notifier.util.BodyFactory;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This is the current Rollbar notifier and the main starting point to send the data to Rollbar.
+ */
 public class Rollbar {
+
+  private static volatile Rollbar notifier;
+
+  private Config config;
+
+  private BodyFactory bodyFactory;
+
+  /**
+   * Constructor.
+   *
+   * @param config the configuration used by the notifier.
+   */
+  public Rollbar(Config config) {
+    this(config, new BodyFactory());
+  }
+
+  private Rollbar(Config config, BodyFactory bodyFactory) {
+    this.config = config;
+    this.bodyFactory = bodyFactory;
+  }
+
+  /**
+   * Method to initialize the library managed notifier instance.
+   *
+   * @param config the configuration.
+   * @return the library managed instance.
+   */
+  public static Rollbar init(Config config) {
+    if (notifier == null) {
+
+      synchronized (Rollbar.class) {
+        if (notifier == null) {
+          notifier = new Rollbar(config);
+        }
+      }
+    }
+
+    return notifier;
+  }
+
+  /**
+   * Handle all uncaught errors on current thread with this `Rollbar`.
+   */
+  public void handleUncaughtErrors() {
+    handleUncaughtErrors(Thread.currentThread());
+  }
+
+  /**
+   * Handle all uncaught errors on {@code thread} with this `Rollbar`.
+   *
+   * @param thread the thread to handle errors on.
+   */
+  public void handleUncaughtErrors(Thread thread) {
```

Comment:
I forgot we had this method, it makes sense to then not have `handleUncaughtErrors` as part of the configuration but instead keep it as an instance method. Well, we could have both, we could have it automatically handle errors for the current thread by default / by configuration, and still provide this method that one can also call if they are managing their own threads.

Reply:
Yes, I think we could have both and introduce the Config flag to auto set this param.

Some pending changes when we finish the implementation are related to the default values in the `ConfigBuilder`, where in the case of the `handleUncaughtErrors` if we finally introduce it we could return `true` by default.